roles/kafka_controller/tasks/get_meta_properties.yml

```diff
@@ -20,9 +20,15 @@
   register: zoo_cluster
   when: kraft_migration|bool
 
-- name: Format Data Directory
-  shell: "{{ binary_base_path }}/bin/kafka-storage  format -t={{ clusterid }} -c {{ kafka_controller.config_file }}  --ignore-formatted"
+- name: Format Data Directory and Create Scram Credentials
+  shell: |
+    {{ binary_base_path }}/bin/kafka-storage format \
+    -t={{ clusterid }} -c {{ kafka_controller.config_file }} \
+    {% if 'SCRAM-SHA-512' in kafka_sasl_enabled_mechanisms or 'SCRAM-SHA-256' in kafka_sasl_enabled_mechanisms %}
```

Comment:
please create a separate task above this task to extract the scram credentials and pass it as an argument here

Reply:
done